### PR TITLE
Moving ClassesTests over to NUnit

### PR DESCRIPTION
**Pull request by sergey-tihon**
part of #7075

### CODE REVIEW

File: tests/fsharp/Compiler/ErrorMessages/ClassesTests.fs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        override __.Bar _ = 1
+            """
+            [|
+                FSharpErrorSeverity.Error, 366, (7, 15, 7, 24), "No implementation was given for those members: \r\n\t'abstract member IOverload.Bar : double -> int'\r\n\t'abstract member IOverload.Bar : int -> int'\r\nNote that all interface members must be implemented and listed under an appropriate 'interface' declaration, e.g. 'interface ... with member ...'."
+                FSharpErrorSeverity.Error, 3213, (8, 21, 8, 24), "The member 'Bar<'a0> : 'a0 -> int' matches multiple overloads of the same method.\nPlease restrict it to one of the following:\r\n   Bar : double -> int\r\n   Bar : int -> int."
+            |]
+
+    [<Test>]
+    let ``Do Cannot Have Visibility Declarations``() =
+        CompilerAssert.ParseWithErrors
```
Comment (sergey-tihon) [inline tests/fsharp/Compiler/ErrorMessages/ClassesTests.fs]: @TIHan You asked for feedback about test framework...
This PR contain first test that call `ParseWithErrors`.

When I can it from VS it opens assertion failed dialog
Win10-1809-1903

If I ignore I see following output
```
Test Name:	Do Cannot Have Visibility Declarations
Test FullName:	FSharp.Compiler.UnitTests.Classes.Do Cannot Have Visibility Declarations
Test Source:	Y:\fsharp\tests\fsharp\Compiler\ErrorMessages\ClassesTests.fs : line 116
Test Outcome:	Failed
Test Duration:	0:00:58.428

Result StackTrace:	
at FSharp.Compiler.Lexhelp.reusingLexbufForParsing[a](LexBuffer`1 lexbuf, FSharpFunc`2 f) in Y:\fsharp\src\fsharp\lexhelp.fs:line 86
   at FSharp.Compiler.Lexhelp.usingLexbufForParsing[a](LexBuffer`1 lexbuf, String filename, FSharpFunc`2 f) in Y:\fsharp\src\fsharp\lexhelp.fs:line 94
   at FSharp.Compiler.SourceCodeServices.ParseAndCheckFile.parseFile(ISourceText sourceText, String fileName, FSharpParsingOptions options, String userOpName, Boolean suggestNamesForErrors) in Y:\fsharp\src\fsharp\service\FSharpCheckerResults.fs:line 1579
   at <StartupCode$FSharp-Compiler-Private>.$Service.ParseFile@415.Invoke(Unit unitVar) in Y:\fsharp\src\fsharp\service\service.fs:line 420
   at Microsoft.FSharp.Control.AsyncPrimitives.CallThenInvoke[T,TResult](AsyncActivation`1 ctxt, TResult result1, FSharpFunc`2 part2) in Y:\fsharp\src\fsharp\FSharp.Core\async.fs:line 416
   at Microsoft.FSharp.Control.Trampoline.Execute(FSharpFunc`2 firstAction) in Y:\fsharp\src\fsharp\FSharp.Core\async.fs:line 113
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.FSharp.Control.AsyncResult`1.Commit() in Y:\fsharp\src\fsharp\FSharp.Core\async.fs:line 368
   at Microsoft.FSharp.Control.AsyncPrimitives.RunSynchronouslyInCurrentThread[a](CancellationToken cancellationToken, FSharpAsync`1 computation) in Y:\fsharp\src\fsharp\FSharp.Core\async.fs:line 913
   at Microsoft.FSharp.Control.AsyncPrimitives.RunSynchronously[T](CancellationToken cancellationToken, FSharpAsync`1 computation, FSharpOption`1 timeout) in Y:\fsharp\src\fsharp\FSharp.Core\async.fs:line 921
   at Microsoft.FSharp.Control.FSharpAsync.RunSynchronously[T](FSharpAsync`1 computation, FSharpOption`1 timeout, FSharpOption`1 cancellationToken) in Y:\fsharp\src\fsharp\FSharp.Core\async.fs:line 1185
   at FSharp.Compiler.UnitTests.CompilerAssert.ParseWithErrors(String source, Tuple`4[] expectedParseErrors) in Y:\fsharp\tests\fsharp\Compiler\CompilerAssert.fs:line 221
   at FSharp.Compiler.UnitTests.Classes.Do Cannot Have Visibility Declarations() in Y:\fsharp\tests\fsharp\Compiler\ErrorMessages\ClassesTests.fs:line 116
Result Message:	
FSharp.Compiler.ErrorLogger+WrappedError : WrappedError(The input array was empty.
Parameter name: array)
Result StandardOutput:	
---------------------------------------------------------------
Executables

CSC                 =C:\Users\sergey\.nuget\packages\Microsoft.Net.Compilers\2.7.0\tools\csc.exe
BUILD_CONFIG        =debug
csc_flags           =/nologo
FSC                 =Y:\fsharp\tests\fsharp\..\..\artifacts\bin\fsc\debug\net472\fsc.exe
fsc_flags           =-r:System.Core.dll --nowarn:20 --define:COMPILED
FSCOREDLLPATH       =Y:\fsharp\tests\fsharp\..\..\artifacts\bin\FSharp.Core\debug\net45\FSharp.Core.dll
FSI                 =Y:\fsharp\tests\fsharp\..\..\artifacts\bin\fsi\debug\net472\fsi.exe
FSIANYCPU           =Y:\fsharp\tests\fsharp\..\..\artifacts\bin\fsiAnyCpu\debug\net472\fsiAnyCpu.exe
fsi_flags           =-r:System.Core.dll --nowarn:20 --define:INTERACTIVE --maxerrors:1 --abortonerror
ILDASM              =C:\Users\sergey\.nuget\packages\runtime.win-x86.Microsoft.NETCore.ILDAsm\2.0.3\runtimes\win-x86\native\ildasm.exe
PEVERIFY            =Y:\fsharp\tests\fsharp\..\..\artifacts\bin\PEVerify\debug\net472\PEVerify.exe
---------------------------------------------------------------


```

Comment (cartermp) [reply tests/fsharp/Compiler/ErrorMessages/ClassesTests.fs]: You can see the same issue in CI here: https://dev.azure.com/dnceng/public/_build/results?buildId=274143&view=ms.vss-test-web.build-test-results-tab&runId=7563620&resultId=100000&paneView=debug

My hunch is that there's an issue with how the compiler assertion function is written.

Comment (TIHan) [reply tests/fsharp/Compiler/ErrorMessages/ClassesTests.fs]: Sorry for the late reply @sergey-tihon .

@KevinRansom made a fix that should resolve the issue.

Also, I made a LOT of changes to CompilerAssert in my default interface method PR that was necessary to test DIMs. So, perhaps that should help. Maybe I'll try to port over those CompilerAssert changes into master.

**Comment by KevinRansom** [comment]
@sergey-tihon, I believe that the assert should now be gone.  So hopefully you are unblocked now.


Thanks

Kevin

**Pull request merged**